ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/query/VaultNamedQueryTransformer.java

```diff
@@ -0,0 +1,38 @@
+package net.corda.v5.ledger.common.query;
+
+import net.corda.v5.base.annotations.Suspendable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+/**
+ * Representation of a transformer function that will be applied to the result set returned by the named query.
+ * <p>
+ * Example usage:
+ * <ul>
+ * <li>Kotlin:<pre>{@code
+ * class MyVaultNamedQueryTransformer : VaultNamedQueryTransformer<ContractState, MyPojo> {
+ *
+ *     override fun transform(state: ContractState, parameters: Map<String, Any>): MyPojo {
+ *         return MyPojo(1)
+ *     }
+ * }
+ * }</pre></li>
+ * <li>Java:<pre>{@code
+ * public class MyVaultNamedQueryTransformer implements VaultNamedQueryTransformer<ContractState, MyPojo> {
+ *
+ *     @Override
+ *     public MyPojo transform(ContractState state, Map<String, Object> parameters) {
+ *         return new MyPojo(1);
+ *     }
+ * }
+ * }</pre></li></ul>
+ *
+ * @param <S> Type of the state returned from the database
+ * @param <T> Type that the original state is transformed into
+ */
+public interface VaultNamedQueryTransformer<S, T> {
+    @Suspendable
+    @NotNull
+    T transform(@NotNull S state, @NotNull Map<String, Object> parameters);
```

Comment:
While I don't disagree with the idea of having common interfaces, for now, we're only concerned with the UTXO ledger and therefore we should put our code there.

Doing this also means we can have `S extends ContractState`.

Can we also rename the generics to `T` for the state and `R` for the return type.